folding/handlers/handlers.py

```diff
@@ -0,0 +1,97 @@
+from abc import ABC, abstractmethod
+from typing import Any 
+from folding.handlers.s3_client import S3Client
+import mimetypes
+import datetime 
+
+class BaseHandler(ABC):
+    """Abstract base class for content handlers.
+    
+    Defines the interface for content handling operations with get/put operations.
+    """
+
+    @abstractmethod
+    def put(self):
+        """Abstract method to store content."""
+        pass
+
+class DigitalOceanS3Handler(BaseHandler):
+    """Handles DigitalOcean Spaces S3 operations for content management.
+
+    Manages file content retrieval and storage operations using DigitalOcean Spaces S3.
+    """
+
+    def __init__(self, bucket_name: str, s3_client: S3Client):
+        """
+        Initializes the handler with a bucet name and an s3 client. 
+
+        Args:
+        bucket_name (str): The name of the s3 bucket to interact with. 
+        s3_client (S3Client): The s3 client to interact with the bucket.
+        """
+
+        self.bucket_name = bucket_name
+        self.s3_client = s3_client
+    
+    def put(
+        self, 
+        file_path: str, 
+        pdb_id: str,
```

Comment:
`pdb_id` is specific to this repo, but what if we wanted to use this S3 handler for SN41? Maybe we can just pass in a location where you want the data to go, rather than the individual pieces